tesla_ver/data_uploading/data_uploading.py

```diff
@@ -0,0 +1,115 @@
+import dash
+import redis
+
+import pandas as pd
+import numpy as np
+import pyarrow as pa
+
+
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+from base64 import b64decode
+from io import StringIO
+
+from tesla_ver.data_uploading.data_uploading_layout import LAYOUT
+from tesla_ver.redis_manager import redis_manager
+
+
+def generate_data_uploading(server):
+    app = dash.Dash(__name__, server=server, url_base_pathname="/datauploading.html/")
+    app.layout = LAYOUT
+
+    @app.callback(
+        [
+            Output("data-table-div", "style"),
+            Output("data-table-visualization", "data"),
+            Output("data-table-visualization", "columns"),
+        ],
+        [Input("upload", "contents"), Input("upload", "filename")],
+    )
+    def upload_data_parsing(contents, filename):
+        def upload_string_to_df(content):
+            """Parse a dash upload string into a single dataframe
+
+                Args:
+                    content (b64 string): string passed from the upload component
+            """
+            _, content_string = content.split(",")
+            decoded = b64decode(content_string)
+            fileish = StringIO(decoded.decode("utf-8"))
+            return pd.read_csv(fileish, sep="\t")
+
+        def parse_multiple_contents(contents, filenames):
+            """parses and merges multiple dataframes into a single selection
+
+            Args:
+                contents (list(b64 encoded string)): list of base64 encoded strings representing dataframes from dash upload
+                filenames (list(str)): list of file names corresponding to each dataframe
+            """
+            pass
+
+        df = pd.DataFrame()
+        if None in [contents, filename]:
+            raise PreventUpdate
+        if len(contents) > 1:
+            parse_multiple_contents(contents, filename)
+        else:
+            df = upload_string_to_df(*contents)
+            df[df < 0] = 0
+            df[~df.isin([np.nan, np.inf, -np.inf]).any(1)]
+            df = df.fillna(0)
```

Comment:
Add comment on why your performing these dataframe operations

Reply:
Especially line 59, not really clear what that line is doing.
`any(1) `may be shorter, but `any(axis=1)` makes it more clear what's happening.

Reply:
addressed in 5a1d5c3